### PR TITLE
My Redox_w layout with full LED support and capslock LED support

### PR DESCRIPTION
**Pull request by pixelbreaker**
My now quite mature layout for redox_w, has been developed over the past year or so for work and gaming.

Also includes my layouts from preonic and tada68

### CODE REVIEW

File: keyboards/tada68/keymaps/pixelbreaker/keymap.c [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  xxx, 	xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 					xxx, 		xxx, \
+  xxx,	 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, 		xxx, \
+  xxx, 	xxx, 		xxx,    							xxx,											xxx, 		xxx, 		MO(_UL), 	xxx, 		xxx, 		xxx),
+};
+
+void tap(uint16_t keycode){
+	register_code(keycode);
+	unregister_code(keycode);
+};
+
```
Comment (zvecr) [inline keyboards/tada68/keymaps/pixelbreaker/keymap.c:147]:
```suggestion

```

File: keyboards/preonic/keymaps/pixelbreaker/config.h [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#ifndef CONFIG_USER_H
+#define CONFIG_USER_H
+
+#include "../../config.h"
+
+#define FORCE_NKRO 1
+
+#endif
```
Comment (zvecr) [inline keyboards/preonic/keymaps/pixelbreaker/config.h:8]:
```suggestion
#pragma once

#define FORCE_NKRO
```

File: keyboards/preonic/keymaps/pixelbreaker/keymap.c [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+// }
+
+// bool music_mask_user(uint16_t keycode) {
+//   switch (keycode) {
+//     case _LOWER:
+//       return false;
+//     default:
+//       return true;
+//   }
+// }
```
Comment (zvecr) [inline keyboards/preonic/keymaps/pixelbreaker/keymap.c:312]:
```suggestion
```

File: keyboards/redox_w/keymaps/pixelbreaker/config.h [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#define TAPPING_TERM 200
```
Comment (zvecr) [inline keyboards/redox_w/keymaps/pixelbreaker/config.h:1]:
```suggestion
#pragma once

#define TAPPING_TERM 200
```

File: keyboards/redox_w/keymaps/pixelbreaker/keymap.c [hunk 5]
```diff
@@ -0,0 +1,306 @@
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
+
```
Comment (zvecr) [inline keyboards/redox_w/keymaps/pixelbreaker/keymap.c:4]:
```suggestion

```

File: keyboards/redox_w/redox_w.c [hunk 6]
```diff
@@ -27,7 +27,7 @@ void matrix_scan_kb(void) {
 }
 
 void led_set_kb(uint8_t usb_led) {
-
+    led_set_user(usb_led);
```
Comment (zvecr) [inline keyboards/redox_w/redox_w.c:30]: We ask nowdays that keyboard changes are submitted separately.
```suggestion

```

File: keyboards/preonic/keymaps/pixelbreaker/rules.mk [hunk 7]
```diff
@@ -0,0 +1,17 @@
+
+
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = yes       # Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
+CONSOLE_ENABLE = no         # Console for debug(+400)
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/pixelbreaker/rules.mk:6]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
```

File: keyboards/redox_w/keymaps/pixelbreaker/config.h [hunk 8]
```diff
@@ -0,0 +1,4 @@
...
+
+#define wht_led_off   PORTD |= (1<<0)
+#define wht_led_on    PORTD &= ~(1<<0)
```
Comment (fauxpark) [inline keyboards/redox_w/keymaps/pixelbreaker/config.h:4]:
```suggestion
#define wht_led_off   writePinHigh(D0)
#define wht_led_on    writePinLow(D0)
```

File: keyboards/tada68/keymaps/pixelbreaker/rules.mk [hunk 9]
```diff
@@ -0,0 +1,23 @@
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = yes       # Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
+CONSOLE_ENABLE = no         # Console for debug(+400)
```
Comment (fauxpark) [inline keyboards/tada68/keymaps/pixelbreaker/rules.mk:8]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
```

File: keyboards/tada68/keymaps/pixelbreaker/rules.mk [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+UNICODE_ENABLE = yes		# Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+COMBO_ENABLE = no 			# Enable key combinations
+TAP_DANCE_ENABLE = yes      # Tappa Dance Bitch
+
+ifndef QUANTUM_DIR
+	include ../../../../Makefile
+endif
```
Comment (fauxpark) [inline keyboards/tada68/keymaps/pixelbreaker/rules.mk:23]:
```suggestion
```

File: keyboards/tada68/keymaps/pixelbreaker/rules.mk [hunk 11]
```diff
@@ -0,0 +1,23 @@
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
+#
```
Comment (fauxpark) [inline keyboards/tada68/keymaps/pixelbreaker/rules.mk:4]:
```suggestion
```

File: keyboards/xd75/keymaps/pixelbreaker/config.h [hunk 12]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2017 Benjamin Kesselring
```
Comment (fauxpark) [inline keyboards/xd75/keymaps/pixelbreaker/config.h:1]: This file can be deleted.

File: keyboards/xd75/keymaps/pixelbreaker/keymap.c [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+#define _QWERTY 0
+#define _SYMB 1
+#define _NAV 2
+#define _FN 3
+#define _TOGGLE 4
+#define _GAMING 5
+#define _GAMING_F 6
+#define _LOCK 7
+#define _UNLOCK 8
+#define _RESET 9
```
Comment (fauxpark) [inline keyboards/xd75/keymaps/pixelbreaker/keymap.c:28]:
```suggestion
enum layer_names {
    _QWERTY,
    _SYMB,
    _NAV,
    _FN,
    _TOGGLE,
    _GAMING,
    _GAMING_F,
    _LOCK,
    _UNLOCK,
    _RESET
};
```

File: keyboards/xd75/keymaps/pixelbreaker/rules.mk [hunk 14]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2013 Jun Wako <wakojun@gmail.com>
```
Comment (fauxpark) [inline keyboards/xd75/keymaps/pixelbreaker/rules.mk:1]: This file can be deleted.

File: keyboards/tada68/keymaps/pixelbreaker/readme.md [hunk 15]
```diff
@@ -0,0 +1,3 @@
+# Maartenwut's layout
```
Comment (fauxpark) [inline keyboards/tada68/keymaps/pixelbreaker/readme.md:1]:
```suggestion
# Pixelbreaker's layout
```
?

File: keyboards/redox_w/keymaps/pixelbreaker/keymap.c [hunk 16]
```diff
@@ -0,0 +1,306 @@
...
+#define _QWERTY 0
+#define _SYMB 1
+#define _NAV 2
+#define _MOUSE 3
+#define _TOGGLE 4
+#define _GAMING 5
+#define _LOCK 6
+#define _UNLOCK 7
+#define _RESET 8
+#define _NUMPAD 9
```
Comment (fauxpark) [inline keyboards/redox_w/keymaps/pixelbreaker/keymap.c:15]:
```suggestion
enum layer_names {
    _QWERTY,
    _SYMB,
    _NAV,
    _MOUSE,
    _TOGGLE,
    _GAMING,
    _LOCK,
    _UNLOCK,
    _RESET,
    _NUMPAD
};
```

File: keyboards/redox_w/keymaps/pixelbreaker/keymap.c [hunk 17]
```diff
@@ -0,0 +1,306 @@
...
+uint16_t anim_frame = 0;
+int anim_dir = 1;
+bool anim_run = true;
+bool locked = false;
+bool locked_light = true;
+
+// Init and switch off the "white" 4th LED
+void led_init_user(void) {
+	DDRD  |= (1<<0) | (1<<1);
+	PORTD |= (1<<0) | (1<<1);
```
Comment (fauxpark) [inline keyboards/redox_w/keymaps/pixelbreaker/keymap.c:67]:
```suggestion
    setPinOutput(D0);
    setPinOutput(D1);
    writePinHigh(D0);
    writePinHigh(D1);
```

File: keyboards/redox_w/keymaps/pixelbreaker/keymap.c [hunk 18]
```diff
@@ -0,0 +1,306 @@
...
+}
+
+// Turn the white LED on/off with CAPSLOCK state
+void led_set_user(uint8_t usb_led) {
+    if (usb_led & (1<<USB_LED_CAPS_LOCK)) {
+        wht_led_on;
+    } else {
+        wht_led_off;
+    }
+}
```
Comment (fauxpark) [inline keyboards/redox_w/keymaps/pixelbreaker/keymap.c:77]:
```suggestion
bool led_update_user(led_t led_state) {
    if (led_state.caps_lock) {
        wht_led_on;
    } else {
        wht_led_off;
    }
    return false;
}

```

File: keyboards/tada68/keymaps/pixelbreaker/keymap.c [hunk 19]
```diff
@@ -0,0 +1,162 @@
+#include QMK_KEYBOARD_H
+
+#define _MA 0
+#define _FL 1
+#define _CL 2
+#define _LO 3
+#define _UL 4
```
Comment (fauxpark) [inline keyboards/tada68/keymaps/pixelbreaker/keymap.c:7]:
```suggestion
enum layer_names {
    _MA,
    _FL,
    _CL,
    _LO,
    _UL
};
```

File: keyboards/tada68/keymaps/pixelbreaker/keymap.c [hunk 20]
```diff
@@ -0,0 +1,162 @@
...
+
+void led_set_user(uint8_t usb_led) {
+	if (usb_led & (1<<USB_LED_CAPS_LOCK)) {
+        // Turn capslock on
+        PORTB |= (1<<6);
+    } else {
+        // Turn capslock off
+        PORTB &= ~(1<<6);
+    }
+}
```
Comment (fauxpark) [inline keyboards/tada68/keymaps/pixelbreaker/keymap.c:156]:
```suggestion
bool led_update_user(led_t led_state) {
	writePin(B6, led_state.caps_lock);
    return false;
}
```

File: keyboards/xd75/keymaps/pixelbreaker/keymap.c [hunk 21]
```diff
@@ -0,0 +1,171 @@
...
+  )
+};
+
+void led_set_user(uint8_t usb_led) {
+    if (usb_led & (1<<USB_LED_CAPS_LOCK)) {
+        capslock_led_on();
+    } else {
+        capslock_led_off();
+    }
+}
```
Comment (fauxpark) [inline keyboards/xd75/keymaps/pixelbreaker/keymap.c:153]:
```suggestion
bool led_update_user(led_t led_state) {
    if (led_state.caps_lock) {
        capslock_led_on();
    } else {
        capslock_led_off();
    }
    return false;
}
```

File: keyboards/tada68/keymaps/pixelbreaker/keymap.c (hunk 1 above)
Comment (pixelbreaker) [reply keyboards/tada68/keymaps/pixelbreaker/keymap.c:147]: What is the difference? 😕

File: keyboards/redox_w/redox_w.c [hunk 22]
```diff
@@ -27,7 +27,7 @@ void matrix_scan_kb(void) {
...
 }
```
Comment (drashna) [inline keyboards/redox_w/redox_w.c:31]:
```suggestion
```

**Pull request closed without merging**